TankClient.java

```diff
@@ -91,442 +109,424 @@ public void framPaint(Graphics graphics) {
-			jmi7.addActionListener(this);
-			jmi7.setActionCommand("level2");
-			jmi8.addActionListener(this);
-			jmi8.setActionCommand("level3");
-			jmi9.addActionListener(this);
-			jmi9.setActionCommand("level4");
-			jmi10.addActionListener(this);
-			jmi10.setActionCommand("Player2");
-			jmi11.addActionListener(this);
-			jmi11.setActionCommand("Join");
-
-			this.setMenuBar(jmb);
-			this.setVisible(true);
+		for (int i = 0; i < metalWall.size(); i++) {
+			MetalWall mw = metalWall.get(i);
+			mw.draw(graphics);
+		}
 
-			for (int i = 0; i < 10; i++) {
-				if (i < 4)
-					homeWall.add(new CommonWall(350, 580 - 21 * i, this));
-				else if (i < 7)
-					homeWall.add(new CommonWall(372 + 22 * (i - 4), 517, this));
-				else
-					homeWall.add(new CommonWall(416, 538 + (i - 7) * 21, this));
+		homeTank.collideWithTanks(tanks);
+		homeTank.collideHome(home);
+		if (player2) {homeTank2.collideWithTanks(tanks);
+		homeTank2.collideHome(home);}
 
-			}
+		for (int i = 0; i < metalWall.size(); i++) {
+			MetalWall w = metalWall.get(i);
+			homeTank.collideWithWall(w);
+			if (player2)homeTank2.collideWithWall(w);
+			w.draw(graphics);
+		}
 
-			for (int i = 0; i < 32; i++) {
-				if (i < 16) {
-					otherWall.add(new CommonWall(200 + 21 * i, 300, this));
-					otherWall.add(new CommonWall(500 + 21 * i, 180, this));
-					otherWall.add(new CommonWall(200, 400 + 21 * i, this));
-					otherWall.add(new CommonWall(500, 400 + 21 * i, this));
-				} else if (i < 32) {
-					otherWall.add(new CommonWall(200 + 21 * (i - 16), 320, this));
-					otherWall.add(new CommonWall(500 + 21 * (i - 16), 220, this));
-					otherWall.add(new CommonWall(222, 400 + 21 * (i - 16), this));
-					otherWall.add(new CommonWall(522, 400 + 21 * (i - 16), this));
-				}
-			}
+		for (int i = 0; i < otherWall.size(); i++) {
+			CommonWall cw = otherWall.get(i);
+			homeTank.collideWithWall(cw);
+			if (player2)homeTank2.collideWithWall(cw);
+			cw.draw(graphics);
+		}
 
-			for (int i = 0; i < 20; i++) {
-				if (i < 10) {
-					metalWall.add(new MetalWall(140 + 30 * i, 150, this));
-					metalWall.add(new MetalWall(600, 400 + 20 * (i), this));
-				} else if (i < 20)
-					metalWall.add(new MetalWall(140 + 30 * (i - 10), 180, this));
+		for (int i = 0; i < homeWall.size(); i++) {
+			CommonWall w = homeWall.get(i);
+			homeTank.collideWithWall(w);
+			if (player2)homeTank2.collideWithWall(w);
+			w.draw(graphics);
+		}
 
-			}
+	}
 
-			for (int i = 0; i < 4; i++) {
-				if (i < 4) {
-					trees.add(new Tree(0 + 30 * i, 360, this));
-					trees.add(new Tree(220 + 30 * i, 360, this));
-					trees.add(new Tree(440 + 30 * i, 360, this));
-					trees.add(new Tree(660 + 30 * i, 360, this));
-				}
+	public TankClient() {
+
+		jmb = new MenuBar();
+		jm1 = new Menu("Game");
+		jm2 = new Menu("Pause/Continue");
+		jm3 = new Menu("Help");
+		jm4 = new Menu("Level");
+		jm5 = new Menu("Addition");
+		jm1.setFont(new Font(TIMES_NEW_ROMAN, Font.BOLD, 15));

```

Comment:
"TIMES NEW ROMAN" galima būtų įdėti kaip private static final string kintamajį ir vietoje to visur įrašyti kintamojo pavadinimą. Tai palengvins kodo palaikomumą, kadangi keičiant šriftą ji reikės daryti tik vienoje vietoje, o ne visur, kur yra šriftas